game/src/game/components/Controllers/AIController.cpp

```diff
@@ -179,11 +200,13 @@ void AIController::HandleMissedCheckpointRespawn(const Timestep& delta_time)
         if (current_checkpoint == -1)
             return;
 
-        transform_->SetPosition(checkpoint_location);
-        FixRespawnOrientation(next_checkpoint_location, checkpoint_location);
-        // add this car's id to respawn, which will be handled by the
-        // gamestateservice
-        game_state_service_->AddRespawnPlayers(this->GetEntity().GetId());
+        vehicle_->Respawn();
+
+        // transform_->SetPosition(checkpoint_location);
+        // FixRespawnOrientation(next_checkpoint_location, checkpoint_location);
+        // // add this car's id to respawn, which will be handled by the
+        // // gamestateservice
+        // game_state_service_->AddRespawnPlayers(this->GetEntity().GetId());
```

Comment:
Can we just remove this instead of commenting out?

Reply:
yessir